Source/WebCore/platform/graphics/controls/PlatformControl.h

```diff
@@ -58,7 +58,7 @@ class PlatformControl {
     virtual void draw(GraphicsContext&, const FloatRoundedRect&, float, const ControlStyle&) { }
 
 protected:
-    ControlPart& m_owningPart;
+    ThreadSafeWeakPtr<ControlPart> m_owningPart;
```

Comment:
Would be nice to add a comment like: `// Cannot be null.` since you're relying on that and the previous code was making it clear(er).